tests/unit/s2n_fingerprint_test.c

```diff
@@ -245,5 +277,440 @@ int main(int argc, char **argv)
         EXPECT_FALSE(s2n_is_grease_value(0x0001));
     };
 
+    /* Test s2n_fingerprint_new / s2n_fingerprint_free */
+    {
+        /* New fails for an invalid handshake method */
+        EXPECT_NULL_WITH_ERRNO(s2n_fingerprint_new(-1), S2N_ERR_INVALID_ARGUMENT);
+
+        /* Free is a no-op for a NULL pointer */
+        EXPECT_SUCCESS(s2n_fingerprint_free(NULL));
+
+        /* New succeeds for a valid handshake method */
+        struct s2n_fingerprint *fingerprint = s2n_fingerprint_new(S2N_FINGERPRINT_JA3);
+        EXPECT_NOT_NULL(fingerprint);
+        EXPECT_EQUAL(fingerprint->method, &ja3_fingerprint);
+        EXPECT_TRUE(s2n_hash_is_ready_for_input(&fingerprint->hash));
+        EXPECT_NULL(fingerprint->client_hello);
+        EXPECT_FALSE(fingerprint->legacy_hash_format);
+        EXPECT_EQUAL(fingerprint->raw_size, 0);
+
+        /* Free cleans up the fingerprint */
+        EXPECT_SUCCESS(s2n_fingerprint_free(&fingerprint));
+        EXPECT_NULL(fingerprint);
+
+        /* Free succeeds again for a freed / NULL fingerprint */
+        EXPECT_SUCCESS(s2n_fingerprint_free(&fingerprint));
+    };
+
+    /* Test s2n_fingerprint_wipe */
+    {
+        /* Safety */
+        EXPECT_FAILURE_WITH_ERRNO(s2n_fingerprint_wipe(NULL), S2N_ERR_INVALID_ARGUMENT);
+
+        /* Initialize an invalid fingerprint with every field set to all 1s */
+        struct s2n_fingerprint fingerprint = { 0 };
+        EXPECT_NOT_NULL(memset(&fingerprint, 1, sizeof(struct s2n_fingerprint)));
+        EXPECT_NOT_NULL(fingerprint.method);
+        EXPECT_TRUE(s2n_hash_is_ready_for_input(&fingerprint.hash));
```

Comment:
Not sure if this is really a concern, but is this potentially making an unnecessary assumption about what `s2n_hash_is_ready_for_input` should be doing? Like, what if `s2n_hash_is_ready_for_input` wanted to call some function pointer in the `s2n_hash_state` or something? Could this test potentially start behaving strangely?

For the purpose of the test I wonder if it would be better to just check the field directly:
```suggestion
        EXPECT_TRUE(fingerprint.hash.is_ready_for_input);
```